src/charm.py

```diff
@@ -424,6 +453,83 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
                 f"Migration of sharding components not permitted, revert config role to {self.role}"
             )
 
+    def __fix_users_for_reuse(self, direct_mongo: MongoDBConnection) -> None:
+        """Fix the users in the DB for storage reuse."""
+        users = direct_mongo.get_all_users()
+        breakpoint()
+        # Update operator password since operator is present.
+        if self.unit.is_leader() and OperatorUser.get_username() in users:
+            logger.info("[Recovery] Update operator password (leader).")
+            direct_mongo.set_user_password(
+                OperatorUser.get_username(), direct_mongo.config.password
+            )
+        # Operator user not present in users, need to create it.
+        elif self.unit.is_leader():
+            logger.info("[Recovery] Create missing operator user (leader).")
+            direct_mongo.create_user(self.mongodb_config)
+        # Not leader and operator present, drop it.
+        elif OperatorUser.get_username() in users:
+            logger.info("[Recovering] Dropping operator user (not leader).")
+            direct_mongo.drop_user(OperatorUser.get_username())
+        # Drop backup and monitor user if present.
+        logger.info("[Recovering] Dropping users backup and monitor.")
+        if BackupUser.get_username() in users:
+            direct_mongo.drop_user(BackupUser.get_username())
+        if MonitorUser.get_username() in users:
+            direct_mongo.drop_user(MonitorUser.get_username())
+        pass
+
+    def _fix_mongodb_for_reuse(self) -> None:
+        """If we are reusing storage, we need to fix multiple things in the mongodb data.
+
+        This is done before starting (so with a downtime).
+        We need to:
+         * Start mongod in degraded mode (no auth validation, no replicaset
+         specified, bind only to local IP).
+         * Update operator password (on leader) and delete it on other units.
+         * Remove backup and monitor users on all units.
+         * Delete the local database (contains the replica set local information) on all units.
+        """
+        logger.debug("Fixing MongoDB for reuse.")
+        self.status.set_and_share_status(MaintenanceStatus("Updating MongoDB for storage reuse."))
```

Comment:
```suggestion
        self.status.set_and_share_status(MaintenanceStatus("Reusing provided storage."))
```

reason: status messages should be abstract